kore/src/Kore/Step/Simplification/OverloadSimplifier.hs

```diff
@@ -62,6 +65,34 @@ data OverloadSimplifier =
         argument, if such an overload exists. Also returns the appropriate
         injection, if necessary.
         -}
+        , getOverloadedWithinSort
+            :: Inj ()
+            -> Symbol
+            -> Sort
+            -> Either String (Maybe (Symbol, Maybe (Inj ())))
+        {- ^Find a symbol overloaded by the argument with the result sort being
+        a subsort of the argument, if a maximum (w.r.t. overloading) such
+        symbol exists. Also returns the appropriate injection, if necessary.
+        -}
+        , unifyOverloadWithSortWithinBound
+            :: Symbol
+            -> Inj ()
+            -> Either String (Maybe (Pair (Symbol, Maybe (Inj ()))))
+        {- Given symbol firstHead of sort S1 and an injection ind{S2, injTo}
+          If there exists a (unique) maximum overloaded symbol secondHead
+          with its result sort S2' within S2, such that there exists a common
+          overload headUnion of firstHead and secondHead
+          with its result sort S within injTo, then return a 'Pair' of the form
+
+            @((headUnion, inj{S, injTo}), (secondHead, inj{S2', S2}))@
+
+          where the injections can miss if they are identities.
+
+          If there are no overloaded symbols candidates, return @Right Nothing@
+
+          If there are candidates, but no maximum, throw an
+          'errorAmbiguousOverloads' with the candidates found
+        -}
```

Comment:
I'm really struggling to understand what the return value of this function means. Maybe it would help to make the return type a custom data type?